### DISCUSSION TITLE
Aula Exibindo Validações

### DISCUSSION THREAD

**Discussion by pedrovazfontes** [balta-io/forum#201] [Tira Dúvidas] [answered]
Pessoal, boa noite!
Eu de novo rs, estou com um problema nesta aula na parte de mostrar as validações.
No minuto 1:30 quando definido a tag small, ele da um erro que não consigo prosseguir.
Ja revisei o código, não sei se deixei algo passar...
Texto do erro:
"src/app/app.component.html:1:7 - error TS2339: Property 'title' does not exist on type 'AppComponent'.

1 <h1>{{title}}</h1>
        ~~~~~

  src/app/app.component.ts:9:16
    9   templateUrl: './app.component.html', // o template pode ser tanto através de uma url, onde será importado um arquivo externo
                     ~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component AppComponent.

ERROR

src/app/app.component.html:18:37 - error TS4111: Property 'title' comes from an index signature, so it must be accessed with ['title'].

18         <small *ngIf="form.controls.title.invalid">Este campo deve conter entre 3 e 60 caracteres</small>
                                       ~~~~~

  src/app/app.component.ts:9:16
    9   templateUrl: './app.component.html', // o template pode ser tanto através de uma url, onde será importado um arquivo externo
                     ~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component AppComponent.
"

em anexo o código html e o component.ts
![app component](https://user-images.githubusercontent.com/61070351/210287196-709f7bc9-852e-4499-94c9-c22a96a95ecc.png)
![html](https://user-images.githubusercontent.com/61070351/210287197-dcca529c-dab9-422f-9969-06301958dac3.png)

**Comment by andrebaltieri** [comment, marked as answer]
Boa noite, @pedrovazfontes, como vai?

Acredito que sua versão do Angular está diferente da do curso. Recomendo que utilize a mesma versão para não ter problemas.

Em relação ao erro, nas novas versões do Angular, as validações são cumulativas no objeto, ou seja, o `title` recebe um **array** e não apenas um valor.

Desta forma, como resultado final, o código fica assim:
```html
<small *ngIf="form.controls.errors?.['invalid']">Este campo deve...</small>
```

**Comment by pedrovazfontes** [comment]
Boa noite, 
fiz a desinstalação e até apaguei as pastas npm da app data(roaming)
Fiz a instalação da versão usada no curso a [11.7.0](https://nodejs.org/en/blog/release/v11.7.0/), porém ainda continua dando problema
segue o novo erro
![erro](https://user-images.githubusercontent.com/61070351/210445611-ce9ae494-a1e6-4ce7-bc9d-5d39c8b4e3a0.png)

**Comment by pedrovazfontes** [comment]
Em tempo, 
Instalei uma versão um pouco mais nova a 16.13 LTS
Ela roda certinho o problema é quando adiciono a linha da validação do <small> ela para de funcionar...
irei prosseguir sem ela para não atrasar os conteúdos
